src/components/hook.js

```diff
@@ -46,23 +47,44 @@ export function createHook(Store, { selector } = {}) {
       }
     };
 
+    const unsubRef = useRef();
+
+    const registerUpdateFn = () => {
+      props && props;
+
+      if (unsubRef.current) {
+        unsubRef.current();
+        unsubRef.current = null;
+      }
+
+      // we call the current ref fn so state is fresh
+      const onUpdate = (...args) => onUpdateRef.current(...args);
+      const unsubscribe = storeState.subscribe(onUpdate);
+      unsubRef.current = unsubscribe;
+    };
+
+    useState(registerUpdateFn);
+
     // if we detect that state has changed, we shedule an immediate re-render
     // (as suggested by react docs https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops)
     // still, it feels silly
     if (prevState !== currentState) {
       setState(currentState);
     }
 
+    if (prevStore !== storeState) {
+      registerUpdateFn();
+      setStore(storeState);
+    }
+
     useIsomorphicLayoutEffect(() => {
-      // we call the current ref fn so state is fresh
-      const onUpdate = (...args) => onUpdateRef.current(...args);
-      // after component is mounted or store changed, we subscribe
-      const unsubscribe = storeState.subscribe(onUpdate);
       return () => {
-        // fired on unmount or every time store changes
-        unsubscribe();
+        if (unsubRef.current) {
+          unsubRef.current();
+          unsubRef.current = null;
+        }
       };
-    }, [storeState]);
+    }, []);
```

Comment:
This make sure it only get called during unmount.

Reply:
why you called @kiddkai ?

Reply:
https://en.wikipedia.org/wiki/Jason_Kidd

Best PG ever.